src/reflect/scala/reflect/internal/Types.scala

```diff
@@ -1194,6 +1194,7 @@ trait Types
       else super.safeToString
     override def narrow: Type = this
     override def kind = "ThisType"
+    override def boundSyms = if (sym.isInstanceOf[StubSymbol]) emptySymbolSet else super.boundSyms
```

Comment:
Would you like to keep this override, given the other more fundamental fix? I'd remove it, but either way is fine.